linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -532,15 +533,19 @@ public void remove() {
    * by Enumerable.)
    */
   public static <TSource> Enumerable<TSource> except(
-      Enumerable<TSource> source0, Enumerable<TSource> source1) {
+      Enumerable<TSource> source0, Enumerable<TSource> source1, boolean all) {
     Set<TSource> set = new HashSet<>();
-    source0.into(set);
-    try (Enumerator<TSource> os = source1.enumerator()) {
+    Collection<TSource> resultCollection = all ? HashMultiset.create() : new HashSet<>();
```

Comment:
Is there a source of definition on `except/intersect all` implementation?

Reply:
hi, @amaliujia I'm not quite sure what do you want to ask, and I guess you were asking the implementation of `EnumerableIntersect` and `EnumerableMinus`.
They are are in [EnumerableIntersect](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableIntersect.java#L45) and [EnumerableMinus](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableMinus.java#L45)

I update the logic of these to support `all`.

Reply:
Thank Yanlin. 

I was looking for some preciously definition of the set op "ALL" implementation (like comments). 

Basically either something like:
```
              // Say for Row R, there are m instances on left and n instances on right,
              // INTERSECT ALL outputs MIN(m, n) instances of R.
```

Or 
```
Calcite implementation follows XX DB's implementation. See their doc(link)
```

If there is no such thing I will just read code.

Reply:
I haven't found that.